Userland/Applications/PixelPaint/MainWidget.cpp

```diff
@@ -1061,6 +1097,63 @@ ImageEditor* MainWidget::current_image_editor()
     return verify_cast<PixelPaint::ImageEditor>(m_tab_widget->active_widget());
 }
 
+void MainWidget::update_action_states_after_layer_stack_change()
+{
+    auto layer_count = current_image_editor() ? current_image_editor()->image().layer_count() : 0;
+    if (!layer_count) {
+        m_merge_active_layer_up_action->set_enabled(false);
+        m_merge_active_layer_down_action->set_enabled(false);
+        m_move_active_layer_to_front_action->set_enabled(false);
+        m_move_active_layer_to_back_action->set_enabled(false);
+        m_move_active_layer_up_action->set_enabled(false);
+        m_move_active_layer_down_action->set_enabled(false);
+        m_flatten_image_action->set_enabled(false);
+        m_remove_active_layer_action->set_enabled(false);
+        m_merge_visible_layers_action->set_enabled(false);
+        m_flip_active_layer_horizontally_action->set_enabled(false);
+        m_flip_active_layer_vertically_action->set_enabled(false);
+        m_rotate_active_layer_clockwise_action->set_enabled(false);
+        m_rotate_active_layer_counterclockwise_action->set_enabled(false);
+        m_crop_active_layer_to_selection_action->set_enabled(false);
+        m_crop_active_layer_to_content_action->set_enabled(false);
+        m_select_previous_layer_action->set_enabled(false);
+        m_select_next_layer_action->set_enabled(false);
+        m_select_top_layer_action->set_enabled(false);
+        m_select_bottom_layer_action->set_enabled(false);
+        return;
+    }
+
+    auto& editor = *current_image_editor();
+
+    bool active_layer_is_frontmost = editor.active_layer() && &editor.image().layer(layer_count - 1) == editor.active_layer();
+    bool active_layer_is_backmost = editor.active_layer() && &editor.image().layer(0) == editor.active_layer();
+
+    m_merge_active_layer_up_action->set_enabled(layer_count > 1 && !active_layer_is_frontmost);
+    m_merge_active_layer_down_action->set_enabled(layer_count > 1 && !active_layer_is_backmost);
+
+    m_move_active_layer_up_action->set_enabled(layer_count > 1 && !active_layer_is_frontmost);
+    m_move_active_layer_down_action->set_enabled(layer_count > 1 && !active_layer_is_backmost);
+
+    m_move_active_layer_to_front_action->set_enabled(layer_count > 1 && !active_layer_is_frontmost);
+    m_move_active_layer_to_back_action->set_enabled(layer_count > 1 && !active_layer_is_backmost);
+
+    m_flatten_image_action->set_enabled(layer_count >= 2);
+    m_merge_visible_layers_action->set_enabled(layer_count >= 2);
```

Comment:
```suggestion
    m_flatten_image_action->set_enabled(layer_count > 1);
    m_merge_visible_layers_action->set_enabled(layer_count > 1);
```

Maybe a bit nitpicky, but every other check does `> 1` instead of `>= 2` so for consistency use one of those styles (I suggested here `> 1` only because it would change less).